src/libaktualizr/storage/sqlstorage_base.cc

```diff
@@ -61,6 +63,48 @@ std::string SQLStorageBase::getTableSchemaFromDb(const std::string& tablename) {
   return schema.value() + ";";
 }
 
+bool SQLStorageBase::dbMigrateForward(int version_from) {
+  SQLite3Guard db = dbConnection();
+  for (int32_t k = version_from + 1; k <= current_schema_version_; k++) {
+    if (db.exec(schema_migrations_.at(static_cast<size_t>(k)), nullptr, nullptr) != SQLITE_OK) {
```

Comment:
Thoughts:

- for safety, first insert the rollback migrations, then execute the forward migration. This way, there is no way to lose the backward migrations if the migration process was interrupted. Even cleaner would be to use sqlite save points (https://www.sqlite.org/lang_savepoint.html) to nest all this migration process in a big transaction. Not sure if we can still do that and keep compatibility. It might be fine though, need to think a bit more about it
- use `INSERT OR REPLACE`, as we'll probably want to have the more recent versions of these

Reply:
I tried to swap, but it doesn't work. Doesn't work because first we need to migrate to create backward_migrations table, then we can insert data in to it. I also tried  to add save point but PrepareStatement always fails when I add it.

Reply:
Good point for swapping, I missed that detail!

I took the liberty to push a version which should work with save points on top of your PR. Please check if it doesn't break your work.